model/jpa/src/main/java/org/keycloak/models/jpa/RealmAdapter.java

```diff
@@ -1725,15 +1669,13 @@ public AuthenticationFlowModel addAuthenticationFlow(AuthenticationFlowModel mod
     }
 
     @Override
-    public List<AuthenticationExecutionModel> getAuthenticationExecutions(String flowId) {
+    public Stream<AuthenticationExecutionModel> getAuthenticationExecutionsStream(String flowId) {
         AuthenticationFlowEntity flow = em.getReference(AuthenticationFlowEntity.class, flowId);
 
         return flow.getExecutions().stream()
                 .filter(e -> getId().equals(e.getRealm().getId()))
                 .map(this::entityToModel)
-                .sorted(AuthenticationExecutionModel.ExecutionComparator.SINGLETON)
-                .collect(Collectors.collectingAndThen(
-                        Collectors.toList(), Collections::unmodifiableList));
+                .sorted(AuthenticationExecutionModel.ExecutionComparator.SINGLETON);
```

Comment:
I think you are right. I switched to `forEachOrdered` from `forEach` in each usage. One concern which I have is whether the corresponding methods in infinispan/RealmAdapter will preserve the ordering in case the data was cached.

Reply:
Yeah, I was also wondering about that.